self_balancing_tree.py

```diff
@@ -0,0 +1,128 @@
+# Hacker rank problem link: https://www.hackerrank.com/challenges/self-balancing-tree
+
+
+class Node:
+    def __init__(self,data , left=None, right=None):
+        self.data = data
+        self.height = 0

```

Comment:
There are 2 ways of keeping track of height: either you store it here or you recompute it with height() function.

Your code does both and it's unclear which approach you're aiming for. However a general comment is that if you cache the height in the tree (as you're doing by adding this variable), you *have to* keep it consistent.